editor/src/components/editor/actions/actions.tsx

```diff
@@ -5604,31 +5572,66 @@ export function isSendPreviewModel(action: any): action is SendPreviewModel {
   return action != null && (action as SendPreviewModel).action === 'SEND_PREVIEW_MODEL'
 }
 
-function revertFileInProjectContents(
+function saveFileInProjectContents(
   projectContents: ProjectContentTreeRoot,
   filePath: string,
 ): ProjectContentTreeRoot {
   const file = getContentsTreeFileFromString(projectContents, filePath)
   if (file == null) {
     return projectContents
   } else {
-    const updatedProjectContents = addFileToProjectContents(
-      projectContents,
-      filePath,
-      revertFile(file),
-    )
-    return updatedProjectContents
+    return addFileToProjectContents(projectContents, filePath, saveFile(file))
   }
 }
 
-function saveFileInProjectContents(
-  projectContents: ProjectContentTreeRoot,
-  filePath: string,
-): ProjectContentTreeRoot {
-  const file = getContentsTreeFileFromString(projectContents, filePath)
-  if (file == null) {
-    return projectContents
-  } else {
-    return addFileToProjectContents(projectContents, filePath, saveFile(file))
+function insertWithReparentStrategies(
+  editor: EditorState,
+  parentPath: InsertionPath,
+  elementToInsert: { elementPath: ElementPath; pathToReparent: ToReparent },
+  indexPosition: IndexPosition,
+  builtInDependencies: BuiltInDependencies,
+): { updatedEditorState: EditorState; newPath: ElementPath } | null {
+  const outcomeResult = getReparentOutcome(
+    builtInDependencies,
+    editor.projectContents,
+    editor.nodeModules.files,
+    editor.canvas.openFile?.filename,
+    elementToInsert.pathToReparent,
+    parentPath,
+    'always',
+    indexPosition,
+  )
+
+  if (outcomeResult == null) {
+    return null
   }
+
+  const { commands: reparentCommands, newPath } = outcomeResult
+
+  const reparentStrategy = reparentStrategyForPaste(
```

Comment:
I guess `reparentStrategyForPaste` should probably be renamed now. I'm not sure what to. Perhaps something like `reparentStrategyForStaticReparent`? Naming is hard...

Reply:
https://github.com/concrete-utopia/utopia/pull/3642/commits/3f275e2492b58a8d448f576334869917ef09b791 
![image](https://github.com/concrete-utopia/utopia/assets/16385508/eae21740-1263-44cf-81ac-58efa08f89c6)